### ISSUE TITLE
the difference between paper‘s figure  and visualized predictd results 

### ISSUE THREAD

**Issue by yangyangsu29** [bradyz/cross_view_transformers#16]
hi，I run the demo in  cross_view_transformers/scripts/example.ipynb，but  I find the difference between paper‘s figure  and the outputs'result , the figures are as fowllows:

![image](https://user-images.githubusercontent.com/23189201/172120059-bf6a8c63-3539-476b-8e4d-c6a1a791e7e1.png)
![image](https://user-images.githubusercontent.com/23189201/172120134-5453a089-c6f8-47ed-8a13-18a7ed3f7ca7.png)
                                          
in my opinion, if we get  the paper's figure, the  model is a three labels but not a simply binary segmentation  model, I look forward to your answer，thank you.

**Comment by DerrickXuNu** [comment]
I think most of the papers right now have separate models for dynamic objects and static road layout, that's why you can only see a single model here. Just my personal perspective.

**Comment by bradyz** [comment]
Derrick is correct - all of the models in this work were trained for single classes for closer comparison to prior works like Lift-Splat and FIERY

**Comment by gongyan1** [comment]
Thank you for your answers, but I want to ask how to achieve the result graph shown by the author.

**Comment by yangyangsu29** [comment]
> @yangyangsu29 Did you solve the problem now? Thanks.
 training each class （vehicle and driveable area）separately，it can reproduce the paper‘s result indeed；in visualization
，the predict‘s map overlay together as follow：
![road_19](https://user-images.githubusercontent.com/23189201/174002968-53a1161c-a741-4ba7-a2ea-5b3a3bb23aea.jpg)

**Comment by gongyan1** [comment]
@yangyangsu29  
`python3 scripts/train.py \
  +experiment=cvt_nuscenes_vehicle
  data.dataset_dir=/media/datasets/nuscenes \
  data.labels_dir=/media/datasets/cvt_labels_nuscenes
`
Thanks for your reply, please allow me to refine the question further.
You mean that when the above command is executed, only the vehicle-related model is trained, I need to change "+experiment=cvt_nuscenes_vehicle" to "+experiment=cvt_nuscenes_road" to train driveable area, but the visualization is not saved after this command. Therefore, I edit the _log_image function in cross_view_transformer/callbacks/visualization_callback.py to store the visualization feature map by opencv. I would like to ask how to add the two feature maps, and whether the author has implemented it in this repo. In other words, I wish you could be a little more detailed.

**Comment by yangyangsu29** [comment]
> @yangyangsu29 `python3 scripts/train.py \ +experiment=cvt_nuscenes_vehicle data.dataset_dir=/media/datasets/nuscenes \ data.labels_dir=/media/datasets/cvt_labels_nuscenes ` Thanks for your reply, please allow me to refine the question further. You mean that when the above command is executed, only the vehicle-related model is trained, I need to change "+experiment=cvt_nuscenes_vehicle" to "+experiment=cvt_nuscenes_road" to train driveable area, but the visualization is not saved after this command. Therefore, I edit the _log_image function in cross_view_transformer/callbacks/visualization_callback.py to store the visualization feature map by opencv. I would like to ask how to add the two feature maps, and whether the author has implemented it in this repo. In other words, I wish you could be a little more detailed.

yes , change "+experiment=cvt_nuscenes_vehicle" to "+experiment=cvt_nuscenes_road" to train driveable area, then run   ./scripts/example.ipynd(modify ckpt_path and so on) to infer the val datasets, in visualization, you need write less codes to show them in a figure as showed above

**Comment by gongyan1** [comment]
@yangyangsu29 Can you be more detailed about this "in visualization, you need write less codes to show them in a figure as showed above"? Thanks.

**Comment by lzm2275965881** [comment]
> @yangyangsu29 Can you be more detailed about this "in visualization, you need write less codes to show them in a figure as showed above"? Thanks.
Have you found a way to combine the two visualization results?  Please give me some advice，thanks !

**Comment by lzm2275965881** [comment]
@gongyan1   Have you found a way to combine the two visualization results? Please give me some advice，thanks !

**Comment by bradyz** [comment]
Sorry for the delay - try this [notebook](https://github.com/bradyz/cross_view_transformers/blob/master/scripts/stitch.ipynb)  
to visualize merge predictions from two models
